### PR TITLE
Namespace CLI

### PR DESCRIPTION
**Pull request by jackrieck**
Using the Namespace SDK to write the CLI.

Updates:
* String matched the enum's so they parse easily when using the json cli method
* added all methods for the namespace program to the CLI

Questions:

* How to do I push my changes to the npm package?
* I moved the IDL to an `idls` directory within the `js` directory, is this where we want to house them?

### CODE REVIEW

File: js/cli/src/namespace.ts
Comment (long-banana) [inline js/cli/src/namespace.ts:8]: Turbo repo pr allows us to import these from the package now

File: js/cli/src/namespace.ts
Comment (long-banana) [inline js/cli/src/namespace.ts:11]: Convention we've been using is to have system/external modules imported above local modules. Move `fs` import above line 6.

File: js/cli/src/namespace.ts
Comment (long-banana) [inline js/cli/src/namespace.ts:237]: This is missing the `env` argument convention that can be passed in to all the other cli commands.

File: js/cli/src/namespace.ts
Comment (long-banana) [inline js/cli/src/namespace.ts:22]: Why does only this one use `anchor.web3`.

File: js/lib/src/state/namespace.ts
Comment (long-banana) [inline js/lib/src/state/namespace.ts:14]: We're moving to an approach more along the lines of this https://github.com/raindrops-protocol/raindrops/blob/57b1f4a19923a9ff0b919f0373ea10648cd506dc/js/lib/src/state/item.ts#L305, with a `toInstruction` and `fromInstruction`(from hasn't been implemented yet) interface.

File: js/cli/example-configs/namespace/initializeNamespace.json
Comment (long-banana) [inline js/cli/example-configs/namespace/initializeNamespace.json:9]: The intended format of these CLI json configs is to match the format that comes down the wire using the IDL format when specifying enum values.

**Review by long-banana — CHANGES_REQUESTED** [review]

**Comment by long-banana** [comment]
> Questions:
> 
> * How to do I push my changes to the npm package?
This will happen via a CI job once that's in place. Until then, the turbo repo pr made it so publishing the package isn't needed  until a PR is merged to main/master.
> * I moved the IDL to an `idls` directory within the `js` directory, is this where we want to house them?
A directory in the `js` directory is where we want to house them, but the format isn't quite right in this PR. You can reference this branch https://github.com/raindrops-protocol/raindrops/tree/xnft-support/js/lib/src/idls for the structure and IDL file content I'd like to move to.

File: js/cli/src/namespace.ts
Comment (jackrieck) [reply js/cli/src/namespace.ts:237]: I'm not sure what the `env` argument would do in this function, the `rpcUrl` already takes care of which env you would be pointed to.

Comment (jackrieck) [reply js/cli/src/namespace.ts:237]: Also this is an internal function used by the higher up cli commands so it shouldn't break any user flow

File: js/cli/src/namespace.ts
Comment (jackrieck) [reply js/cli/src/namespace.ts:8]: Okay, I need to learn about Turbo, I dont quite understand what you mean, I think since this isn't in master I would need to have a follow up with the path changes?

File: js/lib/src/state/namespace.ts
Comment (jackrieck) [reply js/lib/src/state/namespace.ts:14]: This looks very interesting, I like the approach. I am hesitant to change this in this PR because yours is not merged yet so I would need to pull in your PR for the `Instructable` types in `common`

File: js/cli/example-configs/namespace/initializeNamespace.json
Comment (jackrieck) [reply js/cli/example-configs/namespace/initializeNamespace.json:9]: Cool, mentioned above I like this idea, I was thinking I should wait until your PR is merged then make those changes

**Comment by jackrieck** [comment]
> > Questions:
> > 
> > * How to do I push my changes to the npm package?
> 
> This will happen via a CI job once that's in place. Until then, the turbo repo pr made it so publishing the package isn't needed until a PR is merged to main/master.
> 
> > * I moved the IDL to an `idls` directory within the `js` directory, is this where we want to house them?
> 
> A directory in the `js` directory is where we want to house them, but the format isn't quite right in this PR. You can reference this branch https://github.com/raindrops-protocol/raindrops/tree/xnft-support/js/lib/src/idls for the structure and IDL file content I'd like to move to.

I made this change and pushed it up, had a question though.. I had to manually update the generated IDL to match the convention you showed, is this a new convention that maybe a new version of anchor uses?

File: js/lib/src/state/namespace.ts
Comment (long-banana) [reply js/lib/src/state/namespace.ts:14]: You can just steal the `Instructable` interface from my PR and add it here. I'll deal with the conflicts if you get this PR merged before me.

Comment (jackrieck) [reply js/lib/src/state/namespace.ts:14]: Id prefer to do this in a subsequent PR after yours has been merged and approved, its quite a large change and this PR feels pretty big as is, what do you think?

**Review by long-banana — APPROVED** [review]

**Pull request merged**
